src/coreclr/gc/gc.cpp

```diff
@@ -7123,6 +7145,10 @@ void gc_heap::gc_thread_function ()
             {
                 gradual_decommit_in_progress_p = decommit_step (DECOMMIT_TIME_STEP_MILLISECONDS);
             }
+#ifdef DYNAMIC_HEAP_COUNT
+            // check if we should adjust the number of heaps
+            check_heap_count();
```

Comment:
Note that user threads can actually start executing during the call to check_heap_count. This implies that the more space lock can be taken again.